include/pybind11/attr.h

```diff
@@ -517,23 +517,27 @@ template <size_t Nurse, size_t Patient> struct process_attribute<keep_alive<Nurs
 template <typename... Args> struct process_attributes {
     static void init(const Args&... args, function_record *r) {
         PYBIND11_WORKAROUND_INCORRECT_MSVC_C4100(r);
-        int unused[] = { 0, (process_attribute<typename std::decay<Args>::type>::init(args, r), 0) ... };
-        ignore_unused(unused);
+        PYBIND11_WORKAROUND_INCORRECT_ALL_GCC_UNUSED_BUT_SET_PARAMETER(r);
+        PYBIND11_INT_ARRAY_WORKING_AROUND_MSVC_CLANG_ISSUES(
+            {0, (process_attribute<typename std::decay<Args>::type>::init(args, r), 0)...});
```

Comment:
Could you try this?

```suggestion
        using expander = int[];
        (void) expander{0, ((void) process_attribute<typename std::decay<Args>::type>::init(args, r), 0) ...};
```

That's how I'd expect this to look in C++11 (much nicer in C++17). The extra "void" cast is not needed (the one in front of process_attribute), since that's already void, but it's an extra protection that is commonly in this pattern (if it's not void and has a comma operator overload, it will not behave correctly). This works locally for me in Clang. If it doesn't work, then fine, we can go with what you have. :)